shared/src/main/java/com/couchbase/lite/Dictionary.java

```diff
@@ -356,7 +366,7 @@ public Date getDate(String key) {
      */
     @Override
     public Map<String, Object> toMap() {
-        Map<String, Object> result = map != null ? new HashMap<>(map) : new HashMap<String, Object>();
+        Map<String, Object> result = map != null ? new TreeMap<>(map) : new TreeMap<String, Object>();
```

Comment:
As TreeMap maintains the order of keys, I changed it to TreeMap instead of HashMap. I will switch back to HashMap.